src/Alg/Encryption/AES.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\Alg\Encryption;
+
+use SimpleSAML\XMLSecurity\Backend\OpenSSL;
+use SimpleSAML\XMLSecurity\Constants as C;
+use SimpleSAML\XMLSecurity\Key\SymmetricKey;
+
+/**
+ * Class implementing the AES family of encryption algorithms (both AES-CBC and AES-GCM).
+ *
+ * @package simplesamlphp/xml-security
+ */
+class AES extends AbstractEncryptor
+{
+    /** @var string */
+    protected string $default_backend = OpenSSL::class;
+
+
+    /**
+     * AES constructor.
+     *
+     * @param \SimpleSAML\XMLSecurity\Key\SymmetricKey $key The symmetric key to use.
+     * @param string $algId The identifier of this algorithm.
+     */
+    public function __construct(SymmetricKey $key, string $algId = C::BLOCK_ENC_AES256_GCM)
+    {
+        parent::__construct($key, $algId);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSupportedAlgorithms(): array
+    {
+        return [
```

Comment:
Was 3DES left out intentionally, can we use `array_keys(C::BLOCK_CIPHER_ALGORITHMS)` instead?

Reply:
3DES is... 3DES, not AES 😛

Reply:
I know, but it doesn't mean it's not supported.. You're basically by-passing the blacklist if you leave it out here.